samples/verona/std/builtin/dyn.verona

```diff
@@ -0,0 +1,189 @@
+class Dyn
+{
+  // TODO: non-local?
+  // 0000 = Object pointer, no decode step
+  // 0001 = Ref[T], decode by mask
+  // 00020 = Unit, no decode step
+  // 00021 = Bool, decode by trunc
+  // 00022 = I8, decode by trunc
+  // 00023 = U8, decode by trunc
+  // 00024 = I16, decode by trunc
+  // 00025 = U16, decode by trunc
+  // 00026 = I32, decode by trunc
+  // 00027 = U32, decode by trunc
+  // 00030 = F32, decode by trunc and bitcast
+  // 0007-FFFF = F64, decode by subtracting 7<<48
+  create(): Dyn
+  {
+    Dyn::from_unit()
+  }
+
+  ptr_type(): Dyn ::: ()->i64
+  {
+    :[ret i64 0x0000000000000000]:
+  }
+
+  ref_type(): Dyn ::: ()->i64
+  {
+    :[ret i64 0x0001000000000000]:
+  }
+
+  unit_type(): Dyn ::: ()->i64
+  {
+    :[ret i64 0x0002000000000000]:
+  }
+
+  bool_type(): Dyn ::: ()->i64
+  {
+    :[ret i64 0x0002100000000000]:
+  }
+
+  i32_type(): Dyn ::: ()->i64
+  {
+    :[ret i64 0x0002600000000000]:
+  }
+
+  mask(): Dyn ::: ()->i64
+  {
+    :[ret i64 0xFFFFF00000000000]:
+  }
+
+  mask_ptr(): Dyn ::: ()->i64
+  {
+    :[ret i64 0xFFFF000000000000]:
+  }
+
+  make(self: Dyn, ty: Dyn): Dyn ::: (i64, i64)->i64
+  {
+    let a = :[or i64]: self ty
+    a
+  }
+
+  is(self: Dyn, ty: Dyn): Dyn ::: (i64, i64)->i64
+  {
+    let mask = Dyn::mask()
+    let a = :[and i64]: self mask
+    let b = :[icmp eq i64]: ty a
+    Dyn::from_bool(b)
+  }
+
+  is_ptr(self: Dyn, ty: Dyn): Dyn ::: (i64, i64)->i64
+  {
+    let mask = Dyn::mask_ptr()
+    let a = :[and i64]: self mask
+    let b = :[icmp eq i64]: ty a
+    Dyn::from_bool(b)
+  }
+
+  assert(self: Dyn): () ::: (i64)->void
+  {
+    let a = Dyn::to_bool(self)
+    let b = :[icmp ne i1 1,]: a
+    let c = Dyn::from_bool(b)
```

Comment:
I'm not sure I understand what this is doing.

Reply:
It's negating the input, but it's terribly unclear. I'll rewrite it.